federation/api/install/install.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
```

Comment:
I think maintaining the package structure is super confusing, but I guess I
won't stop you. I wlll try to review and note things that shouldn't be
copied on account of pkg/api being non-standard/special.

On Thu, May 12, 2016 at 10:32 AM, Jianhui Z <notifications@github.com>
wrote:

> In federation/api/install/install.go
> <https://github.com/kubernetes/kubernetes/pull/25232#discussion_r63062434>
> :
>
> > @@ -0,0 +1,160 @@
> > +/*
> > +Copyright 2016 The Kubernetes Authors All rights reserved.
> > +
> > +Licensed under the Apache License, Version 2.0 (the "License");
> > +you may not use this file except in compliance with the License.
> > +You may obtain a copy of the License at
> > +
> > +    http://www.apache.org/licenses/LICENSE-2.0
> > +
> > +Unless required by applicable law or agreed to in writing, software
>
> as we will add more k8s objects here, I believe keep the same directory
> structure would help for future work.
> Although I still fell unnecessary of registering only service object for
> the efforts it took and will take.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/25232/files/f3b5aa28612a878b8c2eba79d8a4cf559db4bda3#r63062434>
>